src/core/json/driver.cc

```diff
@@ -4,13 +4,75 @@
 
 #include "src/core/json/driver.h"
 
+#include <absl/strings/str_cat.h>
+
 #include "base/logging.h"
 #include "src/core/json/lexer_impl.h"
 
 using namespace std;
 
 namespace dfly::json {
 
+namespace {
+
+class MaxImpl : public AggFunction {
+  bool ApplyImpl(const JsonType& src) final {
+    if (!src.is_number()) {
+      return false;
+    }
+    val_ = max(val_, src.as_double());
+
+    return true;
+  }
```

Comment:
why would it simplify this code?

Reply:
because you wouldn't need a whole class for a trivial operation, i.e.

```c++
void FMax(It begin, It end) {
  optional<double> maxv;
  for (const auto& val: {begin, end}) {
    if (val.type == number) 
     maxv = max(maxv.value_or(-inf), val.double());
  }
  return maxv ? *maxv : null;
}
```

Reply:
Add it to your TODO list :)